gcp/workers/linter/linter.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env python3
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Run osv-linter on all OSV records."""
+
+import json
+import logging
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+import zipfile
+
+from google.cloud import ndb, storage
+
+import osv.logs
+from osv.models import utcnow
+
+# The Go binary is copied to /usr/local/bin in the Dockerfile
+OSV_LINTER_PATH = '/usr/local/bin/osv-linter'
+VULN_BUCKET = 'osv-test-vulnerabilities'
+ZIP_FILE_PATH = 'all.zip'
+TEST_DATA = './test_data'
```

Comment:
this seems dangerous since we don't know where the CWD is. I would suggest adding a `--work_dir` arg.